### ISSUE TITLE
Blurry icons

### ISSUE THREAD

**Issue by variar** [variar/klogg#97]
BTW did you mess with the Windows icon (top-left of the window and in the taskbar)?
because it looks ugly now. (a bit stretched horizontally, and blurry)

_Originally posted by @wiz0u in https://github.com/variar/klogg/issues/83#issuecomment-529927265_

**Comment by variar** [comment]
> I enjoy doing icons for my apps sometimes. I can make one for you (if you promised my art won't > be deformed at standard 16x16 and 32x32 sizes 😃)

@wiz0u Creating new icon for klogg would be very much appreciated. I'll try my best to use it without deformation 😃.

Also some screenshot with bad icon could help to better see the problem

**Comment by wiz0u** [comment]
Actually I see it's not new with latest builds, it happened already with 19.3.0.393

It seems related to the program being pinned to Taskbar or not.
You can see the issue in the following screenshot
![image](https://user-images.githubusercontent.com/11647984/64695421-5bb3af80-d49c-11e9-96ce-b1342415ca34.png)

My feeling is that when a program is pinned to the Taskbar, the application icon is used (good quality)
But when the program is not pinned, the window icon is used.

You can see the same blurry icon (blurry because it's probably the 16x16 icon enlarged to 32x32) within the Alt-Tab preview

I remember when I was programming for WinAPI in C++ that there was an issue with the WM_SETICON message. By default, only the SMALL icon (16x16) would be loaded as the window icon, and you need to call WM_SETICON a second time, with the LARGE icon (32x32 or more) [and make sure you load the correct version with LoadImage specifying the correct GetSystemMetrics dimensions because LoadIcon would load in such a way that only the small one is available]

**Comment by wiz0u** [comment]
(sorry I don't know the equivalence to this within Qt framework.. it's probably an issue discussed in Qt forums)

**Comment by variar** [comment]
I've found source of this problem. At some point I've "fixed" rc file generation to set klogg version and name in exe-file properties. This generated rc file used ico file with only 32x32 icon. I assume that Windows generates other sizes automagically and downscaling 32x32 icon is never a great idea. I've generated ico-file with all icon sizes from 16 to 48, hope it helps. At least on my machine pinned and unpinned icons are not different anymore.

Should be in build 19.9.0.458
